src/kubernetes/kubernetes.ts

```diff
@@ -616,6 +616,31 @@ export class KubernetesObj extends KubernetesContext {
         console.log("Failure retrieving Namespace " + this.namespace);
         return false;
       });
+
+    // Cancel request after 5 seconds without a response from the readNamespace request.
+    // This usually implies a connectivity issue with OpenShift, which could take a minute or more
+    // before receiving the timeout response.
+    let timeout: NodeJS.Timeout | undefined = undefined;
+    const timeoutPromise: Promise<boolean> = new Promise(resolve => {
+      timeout = setTimeout(() => {
+        vscode.window.showWarningMessage('Connection timed out on "validateNamespaceExists"... Please check your internet/VPN connection.');
```

Comment:
Since this message is displayed to the end user, can you remove the function name from this message? Also do you think it's useful to the user to know where in the validation that it's failing versus just knowing that there's a connectivity issue? Lastly, you can also remove the term `internet/VPN` and reword it to something like this:

"Connection timed out...Please validate the connectivity to OpenShift"

Reply:
Good call, probably not useful for the user to know. More-so for debugging.